tsconfig.json

```diff
@@ -0,0 +1,66 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```

Comment:
Target is what we are building into, yeah? Any reason what ES5 over the default?

Reply:
So many reasons. Actually, we may want to go with ES6 if you care about immutability. (Every browser on Earth supports both ES5 and ES6.)

Reply:
I expected ES6. But actually, does it matter for the target language? Anyway, I'm fine with es6.